### ISSUE TITLE
Move sonatype groupId to org.ankidroid/io.github.ankidroid

### ISSUE THREAD

**Issue by david-allison-1** [david-allison-1/Anki-Android-Backend#33]

**Comment by mikehardy** [comment]
I like the idea of org.ankidroid as sonatype coordinates, but it occurs to me that I am not aware of where that DNS is even controlled and I imagine there are renewal fees etc? :thinking: This may be one for @nicolas-raoul - perhaps he knows the DNS registrar story of ankidroid.org? Without having that on sound footing I would not want to migrate to it.

**Comment by dae** [comment]
Ankitects covers the annual renewal, but isn't this just about having a unique name for the module? AnkiDroid has been using com.ichi2.anki since the early days, and that was never a valid domain (it was probably intended to be net.ichi2.anki)

**Comment by nicolas-raoul** [comment]
Someone recently wanted to add to Sonatype one of my libraries, so I verified ownership, just by creating in the GitHub organization a repository named with the identifier of the Sonatype ticket. Hopefully the same can be done here?
Details at https://issues.sonatype.org/browse/OSSRH-65862 _(the reporter Max Bulankin is the person who added my library to Sonatype)_.

Sorry for the .net/.com mistake 😅. Unfortunately, a Java namespace modification would prevent a lot of users from getting updates. I don't think anyone here is suggesting modifying it anyway.

I would personally recommend transferring the domain as well as the Google Play and social media accounts ownership to [SPI](https://www.spi-inc.org/projects/) (the non-profit that handles Debian's assets) or a similar well-established time-tested free software foundation with a clear and fair workflow that accepts such assets.

**Comment by mikehardy** [comment]
@nicolas-raoul agreed transfer in general of intellectual property / accounts etc to something like SPI is a good idea, but will put that as out of scope for the moment. I'll log a ticket on ankidroid's tracker prior to closing this (or someone else can :smile: )

@dae thanks for the info on ankidroid.org and for the annual fee payment of course !

This is just about a unique name for the module but it seems like a good idea to pick a *good* one before it is fully baked in to automated release processes again because no one I know wants to set up sonatype more than they have to, ever.

It sounds like ankidroid.org at least is under "our" control in that Damien has registrar powers then? Damien - if you confirm you're okay with eventually moving ankidroid.org control to SPI or something (as a future path for community ownership) then proceeding with an org.ankidroid-based coordinate seems fine for this?

And Nicolas - you are right, changing the actual package at this point is a non-starter as it would break the android installers' continuity and users would have to install fresh

**Comment by dae** [comment]
SPI feels like extra red tape to me, but if that's what you guys want then say the word and the domain is yours.

**Comment by mikehardy** [comment]
I certainly don't feel like doing it now, plenty of red tape in my life at the moment, but knowing we have agreement in principle that it *could* happen is important before using it more deeply. As a general statement, I always feel like legal contracts etc in general are annoying wastes of time (red tape) - certainly less useful then people actually trusting each other and behaving honorably - but then every so often they are vital and that's why we have them.

So it or something similar is a thing to do in the future I think such that AnkiDroid moves right along into it's 3rd and 4th decade but not right now while my plate at least is full

I'm +1 on using org.ankidroid coordinates then for any new items, personally. Makes a bunch more sense to me than anything 'ichi2' at this point I think :-)

**Comment by mikehardy** [comment]
just a quick note that my preference would actually be to just have jitpack build this thing correctly and dish the artifact direct from repo configuration rather than have an elaborate (sonatype is kind of elaborate) build/release process with accounts and ownership and everything. The existence of this issue and the points it raises are easy evidence of that. David mentioned in Discord that he had tried jitpack as a build but didn't get very far, while at the same time he is not sure if it was because it's actually not possible or if it was just that he hadn't done it before making it unfamiliar / unsuccessful during the attempt.

So a path forward I think is to do another time-boxed jitpack build attempt, and if that fails, then try get everything set up nicely in sonatype's systems

**Comment by mikehardy** [comment]
jitpack still has not been (re-)tested

In investigated github package repository (relatively new option, vs age of this issue) and it's a non-starter as it [requires auth via a github personal access token for package install](https://docs.github.com/en/packages/learn-github-packages/installing-a-package) (would be fine for publish, but install, no)

**Comment by mikehardy** [comment]
I'll commit to resolving this short-term and re-pointing the ankidroid dependency in order to unblock publish of #200 which will unblock https://github.com/ankidroid/Anki-Android/pull/11599

**Comment by mikehardy** [comment]
Okay, final resolution on path forward:

- jitpack no: we need cross compilation (which transitively implies docker) and jitpack won't to do that, so that's out.
- github package repo: we want easy download and github package repo requires a github personal access token, so no

That leaves setting up Sonatype for the org at org.ankidroid coordinates, so that's the path forward here. I'll start the process. Haven't done it in forever so might take a little bit and if I recall it requires opening a ticket and human interaction on their side, so we'll see.

**Comment by mikehardy** [comment]
as a stopgap I do have personal permissions to the existing coordinates on sonatype and was at least able to publish 0.1.11 that way so this is not a blocker but better to do it sooner rather than later I suppose, or at minimum automate it

**Comment by mikehardy** [comment]
Going to tie this one to the general need to switch ankidroid.org DNS providers as that will make verification of any new sonatype account much easier
